packages/plans-grid-next/src/types.ts

```diff
@@ -153,7 +153,8 @@ export type GridContextProps = {
 	intent?: PlansIntent;
 	siteId?: number | null;
 	useCheckPlanAvailabilityForPurchase: Plans.UseCheckPlanAvailabilityForPurchase;
-	useActionCallback: UseActionCallback;
+	// TODO: Fix this type
+	useAction: () => void;
```

Comment:
I am still studying the changes, but also been thinking a bit generally about this. I wonder if we can avoid a lot of the internal context regarding free trial cta/text/etc by working out this type a little e.g. having primary/secondary actions:

```
interface Action {
    primary: { callback, text, status };
    secondary: { callback, text, status };
    postButtonText;
    etc.
}
```

So internally, we'd simply check and render the "multiple actions" section if both are provided, unconditionally. I'm guessing this will simplify the code a bit 🤔

I'll come back with more after I spend a bit more time with the actions file

Reply:
I like the idea of a primary / secondary in the `Action` interface. Gonna give it more thought as well